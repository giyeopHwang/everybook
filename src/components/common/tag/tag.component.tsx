import { TagTypes } from './tag.const';
import { StyledTag } from './tag.styles';

type TagProps = {
  children: React.ReactNode;
  type?: TagTypes;
};

const Tag = ({ children, type = TagTypes.normal }: TagProps) => {
  return <StyledTag $type={type}>{children}</StyledTag>;
};

export default Tag;
