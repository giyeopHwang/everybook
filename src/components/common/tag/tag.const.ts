export enum TagTypes {
  normal,
  on,
  off,
}
