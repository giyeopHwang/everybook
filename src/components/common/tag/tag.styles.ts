import styled from 'styled-components';
import { TagTypes } from './tag.const';

type TagProps = {
  $type?: TagTypes;
};

export const StyledTag = styled.span<TagProps>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 2rem;
  height: 1.2rem;
  padding: 0.1rem;
  color: white;
  background-color: ${({ $type }) => {
    switch ($type) {
      case TagTypes.on:
        return 'var(--theme-light-tag-highlight)';
      case TagTypes.off:
      case TagTypes.normal:
      default:
        return 'var(--theme-light-tag-default)';
    }
  }};
  font-size: 0.8rem;
`;
