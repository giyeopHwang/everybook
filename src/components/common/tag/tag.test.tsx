import { render } from '@testing-library/react';
import Tag from './tag.component';
import { TagTypes } from './tag.const';

describe('Tag', () => {
  it('renders a untyped Tag component', () => {
    const untypedTag = render(<Tag>UNTYPED</Tag>);
    expect(untypedTag).toMatchSnapshot();
  });

  it('renders a normal-typed Tag component', () => {
    const normalTag = render(<Tag type={TagTypes.normal}>NORMAL</Tag>);
    expect(normalTag).toMatchSnapshot();
  });

  it('renders a on-typed Tag component', () => {
    const onTag = render(<Tag type={TagTypes.on}>ON</Tag>);
    expect(onTag).toMatchSnapshot();
  });

  it('renders a off-typed Tag component', () => {
    const onTag = render(<Tag type={TagTypes.off}>OFF</Tag>);
    expect(onTag).toMatchSnapshot();
  });
});
