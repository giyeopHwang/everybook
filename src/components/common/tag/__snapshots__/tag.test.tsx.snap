// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tag > renders a normal-typed Tag component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXGCL kjVut"
      >
        NORMAL
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXGCL kjVut"
    >
      NORMAL
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag > renders a off-typed Tag component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXGCL kjVut"
      >
        OFF
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXGCL kjVut"
    >
      OFF
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag > renders a on-typed Tag component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXGCL bOrwve"
      >
        ON
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXGCL bOrwve"
    >
      ON
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tag > renders a untyped Tag component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="sc-bcXGCL kjVut"
      >
        UNTYPED
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="sc-bcXGCL kjVut"
    >
      UNTYPED
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
